src/posix/platform/infra_if.cpp

```diff
@@ -276,7 +280,14 @@ otError InfraNetif::SendIcmp6Nd(uint32_t            aInfraIfIndex,
     return error;
 }
 
-bool InfraNetif::IsRunning(void) const { return (GetFlags() & IFF_RUNNING) && HasLinkLocalAddress(); }
+bool InfraNetif::IsRunning(void) const
+{
+    if (!mInfraIfIndex)
+    {
+        return false;
+    }
+    return (GetFlags() & IFF_RUNNING) && HasLinkLocalAddress();
```

Comment:
Should we use single `return`? I think we have in style guide?

Reply:
Oops. I somehow forgot about this rule when writing this part of code.